### PR TITLE
Support Gradle 5.6 Debugging API

### PR DESCRIPTION
**Pull request by drodriguezalfayate**
We check if Gradle supports debugOptions or not.

* If gradle supports debugOptions (at least Gradle 5.6),  new debugOptions specification is used, delegating debugging process on Gradle JavaExecSpec.
* If gradle doesn't support debugOptions (before Gradle 5.6), previous functionality is maintained (although, perhaps it should be rewritten to use ```-agentlib:jdwp```, I haven't made this change, since it works properly, and it could introduce problems in different installations)

This issue is described in https://youtrack.jetbrains.com/issue/IDEA-226922, and proposed change allows to debug application properly

### CODE REVIEW

**Comment by boris-petrov** [comment]
Thanks for this, I left a question/comment inline.

**Comment by drodriguezalfayate** [comment]
Boris,

Excuse me, I can't see the question/comment....

**Comment by boris-petrov** [comment]
@drodriguezalfayate - ah, sorry, I hadn't pressed "submit". I added it.

File: libs/gretty/src/main/groovy/org/akhikhl/gretty/DefaultLauncher.groovy [hunk 1]
```diff
@@ -101,12 +101,31 @@ class DefaultLauncher extends LauncherBase {
...
+
       spec.classpath = project.files(runnerClasspath)
       def jvmArgs = params.jvmArgs
+
       if(params.debug) {
-        jvmArgs.add '-Xdebug'
-        String debugArg = "-Xrunjdwp:transport=dt_socket,server=y,suspend=${params.debugSuspend ? 'y' : 'n'},address=${params.debugPort}"
-        jvmArgs.add debugArg
+        // Checking if we are in pre 5.6 Gradle API
+        if(spec.metaClass.getMetaProperty('debugOptions')==null) {
```
[the remark this answers is not in the document]
Comment (drodriguezalfayate) [reply libs/gretty/src/main/groovy/org/akhikhl/gretty/DefaultLauncher.groovy:110]: I don't like it very much also...

I tried using [project.getGradle().getGradleVersion()](https://docs.gradle.org/current/javadoc/org/gradle/api/invocation/Gradle.html#getGradleVersion--), the problem with that approach is that it returns a `String`, so accessing major and minor version requires parsing the String. With this approach verifying if this option is usable, you have to check if `major.minor` fulfills the following rules:

* Major is greater or equal than `6`
* Or major is `5` and minor is greater or equal than `6`

This is not inherently bad, but you have to make assumptions about how Gradle version is made up (nowadays is `major.minor.patchset`), but you don't have certainty if it'll be always this way, or if in a custom gradle installation gradle versioning could have been changed.

Anyway if you think that is better to parse the string, I don't have any problem to make the changes.

**Pull request merged**